ckanext/saml2/plugin.py

```diff
@@ -211,6 +213,55 @@ def saml2_user_delete(context, data_dict):
     ckan_user_delete(context, data_dict)
 
 
+def saml2_check_for_user_update(id):
+    c = p.toolkit.c
+    user_info = saml2_get_user_info(id)
+    if user_info is not None:
+        c.allow_user_changes = p.toolkit.asbool(
+            config.get('ckan.saml2.allow_user_changes', False))
+        c.is_allow_update = user_info[0].allow_update
+    else:
+        c.allow_user_changes = c.is_allow_update = None
+
+
+def saml2_user_update(context, data_dict):
+    id = logic.get_or_bust(data_dict, 'id')
+    c = p.toolkit.c
+    saml2_check_for_user_update(id)
+    name_id = saml2_get_user_name_id(id)
+    if name_id is not None:
+        if c.allow_user_changes:
+            allow_update_params = data_dict.get('allow_update')
+            if allow_update_params is not None:
+                allow_update_params = p.toolkit.asbool(data_dict.get('allow_update'))
+                model.Session.query(SAML2User).filter_by(name_id=name_id).\
+                    update({'allow_update': allow_update_params})
+                model.Session.commit()
+                if allow_update_params:
+                    return ckan_user_update(context, data_dict)
+            elif not isinstance(c.allow_update_checkbox, str):
+                model.Session.query(SAML2User).filter_by(name_id=name_id).\
+                    update({'allow_update': c.allow_update_checkbox})
+                model.Session.commit()
+                if c.allow_update_checkbox:
+                    return ckan_user_update(context, data_dict)
+            if 'api_version' not in context:
+                h.redirect_to('user_datasets', id=id)
+        else:
```

Comment:
Wouldn't it be better to raise error as soon as possible, instead of all this checks? Even if user send the same values to update as he has now, he is not allowed to do this